### PR TITLE
Rename labelpattern to bibtexkeypattern and fix pref issue

### PR DESCRIPTION
**Pull request by matthiasgeiger**
As mentioned by @lenhard in https://github.com/JabRef/jabref/pull/1626#issuecomment-238533144 - this is a follow up PR:

- [x] Rename label pattern to bibtex key pattern.
  - [x] rename Help file reference (at least until 3.6 is released)
  - [x] rename prefs key 
- [x] Integrate label pattern preferences tree into global preferences tree
- [x] Possible migration of old key patterns (The first time JabRef is run it is checked whether old patterns are there - if yes, they are migrated. Subsequently migration is skipped.)
- ~~Change in CHANGELOG.md described~~ internal cleanup
- ~~Tests created for changes~~ old tests still pass
- ~~Screenshots added (for bigger UI changes)~~ no UI changes
- [x] Manually tested changed features in running JabRef
- [x] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
  - [x] duplicate old Page under new name (at least until 3.6 is released) https://github.com/JabRef/help.jabref.org/issues/45

### CODE REVIEW

**Comment by matthiasgeiger** [comment]
@JabRef/developers Two related questions regarding the migration of LabelPatterns: 
- What happens if old prefs after migration? Delete them?
- What happens if old and new prefs are there? Keep the new ones?

**Comment by koppor** [comment]
Keep the old prefs. They are not consuming much space in the registry. If someone starts an old JabRef version, the old patterns should still be there. It might be that users update from <= 3.5 to a far newer version (4.0?) and want to go back to 3.5, because something does not work on 4.0.

Keep the new ones, if old prefs are there.

I foresee that some users will complain that the patterns change when having used an old version and changed the patterns there. But I think, those guys will be seldom :)

**Comment by stefan-kolb** [comment]
Probably just keeping them is the easiest alternative, as long as this does not create any side effects.

**Comment by matthiasgeiger** [comment]
Chosen approach is: The first time JabRef is run it is checked whether old patterns are there - if yes, they are migrated. Subsequently migration is skipped. Old prefs are not deleted.

Apart from the changes in the help.jabref.org repo this is good to go from my side.

File: src/main/java/net/sf/jabref/migrations/PreferencesMigrations.java [hunk 1]
```diff
@@ -79,4 +88,43 @@ public static void upgradeSortOrder() {
         }
     }
 
+    /**
+     * Migrate LabelPattern configuration from version 3.3-3.5 to new BibtexKeyPatterns
```
Comment (koppor) [inline src/main/java/net/sf/jabref/migrations/PreferencesMigrations.java]: Minor comment: Could we also upgrade from <3.3? We still have users coming from JabRef 2.9.2.

**Comment by koppor** [comment]
Besides the minor comment: LGTM.

As far as I can see, import/export of preferences doesn't include the bibtex key patterns, does it?

File: src/main/java/net/sf/jabref/migrations/PreferencesMigrations.java [hunk 2]
```diff
@@ -79,4 +88,43 @@ public static void upgradeSortOrder() {
...
+                // Check whether old defaultLabelPattern is set
+                String oldDefault = mainPrefsNode.get("defaultLabelPattern", null);
+                if(oldDefault!=null) {
+                    prefs.put(JabRefPreferences.DEFAULT_BIBTEX_KEY_PATTERN, oldDefault);
+                    LOGGER.info("Upgraded old default key generator pattern '"+oldDefault+"' to new version.");
+                }
+
+            }
+            if (mainPrefsNode.nodeExists("bibtexkeypatterns")) {
+                return; //Pref node already exists do not migrate from previous version
```
Comment (koppor) [inline src/main/java/net/sf/jabref/migrations/PreferencesMigrations.java]: Could you move the comment above the statement?

File: src/main/java/net/sf/jabref/migrations/PreferencesMigrations.java (hunk 1 above)
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/migrations/PreferencesMigrations.java]: Should be possible. I'll implement the logic that the newest available prefs are migrated to 3.6

**Comment by matthiasgeiger** [comment]
Prefs were already reported correctly, regarding this point I was wrong at the Dev call. But I should check whether importing is now working fine.

**Comment by stefan-kolb** [comment]
LGTM :+1:

**Comment by matthiasgeiger** [comment]
Added further migrations fors 3.0-3.2 and <3.0.

Exporting and importing prefs is working with KeyPatterns.

**Comment by oscargus** [comment]
Does this fix #1257 ? (Except the 3.2 to 3.3 part...) In that case add a ChangeLog entry and remove the wontfix tag. :-)

**Comment by matthiasgeiger** [comment]
Added the changelog entry.

... can be merged as soon as https://github.com/JabRef/help.jabref.org/pull/46 is merged - which fixes the failing test.

**Pull request merged**